src/Common/Common.Api/Reports/ReportsClient.cs

```diff
@@ -93,5 +103,70 @@ public IEnumerable<Import> GetImportsForWorkspace(Guid workspaceId)
         {
             return this.Client.Imports.GetImports(groupId: workspaceId.ToString()).Value?.Select(x => (Import)x);
         }
+
+        public Guid PostImport(string datasetDisplayName, string filePath)
+        {
+            using (var fileStream = new StreamReader(filePath))
+            {
+                var response = this.Client.Imports.PostImportWithFile(
+                    fileStream: fileStream.BaseStream,
+                    datasetDisplayName: datasetDisplayName,
+                    nameConflict: PowerBI.Api.V2.Models.ImportConflictHandlerMode.CreateOrOverwrite
+                );
+                return Guid.Parse(response.Id);
+            }
+        }
+
+        public Guid PostImportForWorkspace(Guid workspaceId, string datasetDisplayName, string filePath)
+        {
+            using (var fileStream = new StreamReader(filePath))
+            {
+                var response = this.Client.Imports.PostImportWithFileInGroup(
+                    groupId: workspaceId.ToString(),
+                    fileStream: fileStream.BaseStream,
+                    datasetDisplayName: datasetDisplayName,
+                    nameConflict: PowerBI.Api.V2.Models.ImportConflictHandlerMode.CreateOrOverwrite
+                );
+                return Guid.Parse(response.Id);
+            }
+        }
+
+        public Report PostReport(string reportName, string filePath)
+        {
+            var importId = this.PostImport(reportName, filePath);
+
+            Import import = null;
+            do
+            {
+                import = this.GetImport(importId: importId);
+                if (import.ImportState != "Succeeded")
```

Comment:
Stylistically, I think we prefer to always use brackets in conditional statements.